app.py

```diff
@@ -70,11 +130,12 @@ def Items_To_Do():
                 mycollection.delete_one(myquery)
         return redirect("/")
 
-
+    @login_required
     @app.route('/items/create_item_page', methods = ['POST', 'GET'])
     def create_item_page():
         return render_template('CreateCard.html')
 
+    @login_required
     @app.route('/items/Items_To_Add', methods = ['POST', 'GET'])
```

Comment:
None of these routes actually verify that the user has the appropriate role attached. If I log in with a user with read-only permissions, but I then just type the appropriate URL in my browser (`http://localhost:5000/items/create_item_page`) then I can bypass your authorisation. We should explicitly check the users role for appropriate endpoints (as a first pass, we can do that explicitly at the start of each function, but a better solution would be to add a new decorator (e.g. `@write_required`) that does that check for us